File: apps/pos/native-android/app/src/main/java/com/desktopkitchen/pos/models/Reports.kt
package com.desktopkitchen.pos.models

import com.squareup.moshi.JsonClass

@JsonClass(generateAdapter = true)
data class SalesReport(
    val period: String,
    val total_revenue: Double = 0.0,
    val order_count: Int = 0,
    val avg_ticket: Double = 0.0,
    val tip_total: Double = 0.0
)

@JsonClass(generateAdapter = true)
data class TopItemsReport(
    val item_name: String,
    val quantity_sold: Int = 0,
    val revenue: Double = 0.0
)

@JsonClass(generateAdapter = true)
data class HourlyReport(
    val hour: Int,
    val orders: Int = 0,
    val revenue: Double = 0.0,
    val avg_ticket: Double = 0.0
) {
    val hourLabel: String
        get() {
            val h = if (hour % 12 == 0) 12 else hour % 12
            val ampm = if (hour < 12) "AM" else "PM"
            return "$h$ampm"
        }
}

@JsonClass(generateAdapter = true)
data class LiveDashboardKPIs(
    val order_count: Int? = null,
    val revenue: Double? = null,
    val avg_ticket: Double? = null,
    val tips: Double? = null,
    val cash_orders: Int? = null,
    val card_orders: Int? = null,
    val cash_revenue: Double? = null,
    val card_revenue: Double? = null
)

@JsonClass(generateAdapter = true)
data class LiveDashboardData(
    val date: String,
    val kpis: LiveDashboardKPIs,
    val hourly: List<LiveHourlyEntry> = emptyList(),
    val sources: List<LiveSourceEntry> = emptyList(),
    val topItems: List<LiveTopItem> = emptyList()
)

@JsonClass(generateAdapter = true)
data class LiveHourlyEntry(
    val hour: Int,
    val orders: Int = 0,
    val revenue: Double = 0.0
)

@JsonClass(generateAdapter = true)
data class LiveSourceEntry(
    val source: String,
    val count: Int = 0,
    val revenue: Double = 0.0
)

@JsonClass(generateAdapter = true)
data class LiveTopItem(
    val item_name: String,
    val qty: Int = 0
)

File: apps/pos/native-windows/DesktopKitchenPOS/Models/Reports.cs
using System.Text.Json.Serialization;

namespace DesktopKitchenPOS.Models;

public record SalesReport
{
    [JsonPropertyName("period")]
    public string Period { get; init; } = "";

    [JsonPropertyName("total_revenue")]
    public double TotalRevenue { get; init; }

    [JsonPropertyName("order_count")]
    public int OrderCount { get; init; }

    [JsonPropertyName("avg_ticket")]
    public double AvgTicket { get; init; }

    [JsonPropertyName("tip_total")]
    public double TipTotal { get; init; }
}

public record TopItemsReport
{
    [JsonPropertyName("item_name")]
    public string ItemName { get; init; } = "";

    [JsonPropertyName("quantity_sold")]
    public int QuantitySold { get; init; }

    [JsonPropertyName("revenue")]
    public double Revenue { get; init; }
}

public record HourlyReport
{
    [JsonPropertyName("hour")]
    public int Hour { get; init; }

    [JsonPropertyName("orders")]
    public int Orders { get; init; }

    [JsonPropertyName("revenue")]
    public double Revenue { get; init; }

    [JsonPropertyName("avg_ticket")]
    public double AvgTicket { get; init; }

    public string HourLabel
    {
        get
        {
            var h = Hour % 12 == 0 ? 12 : Hour % 12;
            var ampm = Hour < 12 ? "AM" : "PM";
            return $"{h}{ampm}";
        }
    }
}

public record LiveDashboardKPIs
{
    [JsonPropertyName("order_count")]
    public int? OrderCount { get; init; }

    [JsonPropertyName("revenue")]
    public double? Revenue { get; init; }

    [JsonPropertyName("avg_ticket")]
    public double? AvgTicket { get; init; }

    [JsonPropertyName("tips")]
    public double? Tips { get; init; }

    [JsonPropertyName("cash_orders")]
    public int? CashOrders { get; init; }

    [JsonPropertyName("card_orders")]
    public int? CardOrders { get; init; }

    [JsonPropertyName("cash_revenue")]
    public double? CashRevenue { get; init; }

    [JsonPropertyName("card_revenue")]
    public double? CardRevenue { get; init; }
}

public record LiveDashboardData
{
    [JsonPropertyName("date")]
    public string Date { get; init; } = "";

    [JsonPropertyName("kpis")]
    public LiveDashboardKPIs Kpis { get; init; } = new();

    [JsonPropertyName("hourly")]
    public List<LiveHourlyEntry> Hourly { get; init; } = new();

    [JsonPropertyName("sources")]
    public List<LiveSourceEntry> Sources { get; init; } = new();

    [JsonPropertyName("topItems")]
    public List<LiveTopItem> TopItems { get; init; } = new();
}

public record LiveHourlyEntry
{
    [JsonPropertyName("hour")]
    public int Hour { get; init; }

    [JsonPropertyName("orders")]
    public int Orders { get; init; }

    [JsonPropertyName("revenue")]
    public double Revenue { get; init; }
}

public record LiveSourceEntry
{
    [JsonPropertyName("source")]
    public string Source { get; init; } = "";

    [JsonPropertyName("count")]
    public int Count { get; init; }

    [JsonPropertyName("revenue")]
    public double Revenue { get; init; }
}

public record LiveTopItem
{
    [JsonPropertyName("item_name")]
    public string ItemName { get; init; } = "";

    [JsonPropertyName("qty")]
    public int Qty { get; init; }
}
